src/modelmatrix.jl

```diff
@@ -62,7 +62,7 @@ function modelmat_cols(::Type{T},
     ## contrast matrix
     reindex = [findfirst(equalto(l), contrast.levels) for l in CategoricalArrays.index(v.pool)]
     contrastmatrix = convert(T, contrast.matrix)
-    return indexrows(contrastmatrix, reindex[v.refs])
+    return indexrows(contrastmatrix, Vector{Int}(reindex[v.refs]))
```

Comment:
Better directly mark the comprehension as `Int` above?

Reply:
you'd think so but it can contain `nothing`. (if not all levels of the categorical vector are contained in the contrast values). the test that hit this is the one where you make a model frame from a sub-dataframe.

Reply:
in that case you don't even _hit_ the `nothing` values when you index with `v.refs`, and `ContrastMatrix` constructor uses unique values (rather than levels) so this is safe (or rather, an error would be detected earlier).

Reply:
maybe use `coalesce` instead, on the comprehension? I feel like I saw a warning to that effect.

Reply:
`coalesce` wouldn't throw an error in the presence of `nothing`, so a type assertion would be better.

We could also use `indexin`, that could be faster with many levels, but it would require converting the result (though hopefully at some point this kind of conversion will stop copying the underlying data).